src/frontend.md

```diff
@@ -5,31 +5,54 @@
 ```
 src/
   components/ (can be dumb and smart)
-    bankAccount/
+    [All components should be placed in some entity folder or shared folder]
+    bankAccount/ (entity, module, .etc)
       BankAccountTable/
         [no nested directories]
+        [avoid index.ts]
+        [other files should be placed in queries or services]
         BankAccountTable.ts
         BankAccountTable.test.ts
         BankAccountTableBadge.ts (child component)
         BankAccountTable.styled.ts
+        BankAccountTable.hook.ts (ui hook used only in the current component)
+        icon.svg (the icon used only in the current component)
     shared/
       SomeNonEntityComponent/
+      media/ (shared icons)
+        [no nested directories]
+        icon.svg
   pages/
     bankAccount/
       BankAccountList/
+        [no nested directories]
+        [other files should be placed in queries or services]
+        [components for the page should be placed in components folder]
         BankAccountList.page.ts
         BankAccountList.page.test.ts
   queries/
-    bankAccount.query.ts
+    bankAccount.query.ts (query hook)
   services/
     [can be splitted into several files]
-    bankAccount/
+    bankAccount/ (utils, helpers, all the things that should not be put in the components folder)
+      [no nested directories]
       bankAccount.service.ts
       bankAccountJournal.service.ts
+      bankAccount.hook.ts (entity hook)
     [or single file]
     auditLog.service.ts
+    hook.ts (common hook)
 ```
 
+Notes:
+
+- **avoid adding additional folders to the project**;
+- component ui hook [example](https://github.com/OsomePteLtd/agent-factory-tickets/blob/7ca56a50993592d1dbfc62a272559423e0b16303/src/components/reconciliation/shared/Basket/Basket.tsx);
+- query hook [example](https://github.com/OsomePteLtd/agent/blob/main/src/queries/auditLog.query.ts);
+- entity hook [example](https://github.com/OsomePteLtd/websome/blob/main/src/hooks/ticket/useTicketNavigation.ts);
```

Comment:
what does mean `entity` hook? and where should it be placed?

Reply:
I mean a hook used in several components in bankAccount entity